src/native/eventpipe/ds-ipc.c

```diff
@@ -361,12 +383,20 @@ ds_ipc_stream_factory_get_next_available_stream (ds_ipc_error_callback_func call
 			DS_IPC_TIMEOUT_INFINITE :
 			ipc_stream_factory_get_next_timeout (poll_timeout_ms);
 
-		poll_attempts++;
-		DS_LOG_INFO_2 ("ds_ipc_stream_factory_get_next_available_stream - Poll attempt: %d, timeout: %dms.\n", poll_attempts, poll_timeout_ms);
-
-		ipc_log_poll_handles (&ipc_poll_handles);
+		int32_t ret_val;
+		if ( ds_rt_ipc_poll_handle_array_size (&ipc_poll_handles) > 0) {
+			poll_attempts++;
+			DS_LOG_DEBUG_2 ("ds_ipc_stream_factory_get_next_available_stream - Poll attempt: %d, timeout: %dms.", poll_attempts, poll_timeout_ms);
+			ipc_log_poll_handles (&ipc_poll_handles);
+			ret_val = ds_ipc_poll (ds_rt_ipc_poll_handle_array_data (&ipc_poll_handles), ds_rt_ipc_poll_handle_array_size (&ipc_poll_handles), poll_timeout_ms, callback);
+		} else {
+			if (poll_timeout_ms == DS_IPC_TIMEOUT_INFINITE)
+				poll_timeout_ms = DS_IPC_POLL_TIMEOUT_MAX_MS;
+			DS_LOG_DEBUG_1 ("ds_ipc_stream_factory_get_next_available_stream - Nothing to poll, sleeping using timeout: %dms.", poll_timeout_ms);
+			ep_rt_thread_sleep ((uint64_t)poll_timeout_ms * NUM_NANOSECONDS_IN_1_MS);
+			ret_val = 0; // timeout
+		}
```

Comment:
If we have nothing to poll, shouldn't we not ever be calling this logic? I'm assuming this is meant for the case where we've disabled all port types.

Reply:
No, when you disable all port types you won't get here at all. This handles the case when we have ports to connect to but connect fails, then there won't be any handles added to the list and this is needed to correctly sleep the thread or you will spin this loop without any sleep at all, that could for example happen if endpoint is not yet up an running or some other error happening quickly in connect. I have observed this case when using TCP/IP and for example connect to local 127.0.0.1 without any running listener, then `ds_ipc_stream_factory_get_next_available_stream` will spin without interuption.